api/types/access_request.go

```diff
@@ -83,7 +83,7 @@ func NewAccessRequest(name string, user string, roles ...string) (AccessRequest,
 		},
 		Spec: AccessRequestSpecV3{
 			User:  user,
-			Roles: roles,
+			Roles: append([]string{role}, additionalRoles...),
```

Comment:
Add a check for `role == ""`, since this will pass `CheckAndSetDefaults` with `Roles: []string{""}`

Reply:
Request's `CheckAndSetDefaults` needs to modified to take this into account. This applies not only to `role`, but to any additional roles which you might pass in as empty strings.